src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -52,7 +58,20 @@ internal abstract partial class PatternMatcher : IDisposable
             bool allowFuzzyMatching = false)
         {
             culture = culture ?? CultureInfo.CurrentCulture;
+            if (culture.Name == "en-US")
```

Comment:
> Expecting culture-sensitive result on en-US, but getting ordinal result?

Yes.  I find it highly questionable that people would write code that needs culture sensitive results in en-us.  And even if they did, i'm ok with that scenario regressing given the gain here.

> Expecting the same speedup as en-US on a different culture when data is in ASCII range?

I'm also ok with that.  At worst the code performs the same as today for non-en-us.  But for the common case of en-us and english code, we get a boost.